contracts/PFAUnit.sol

```diff
@@ -104,12 +107,24 @@ contract PFAUnit is PFA, ERC721 /* G_NFT */ {
         _tokenURI = tokenURI_;
     }
 
+    /// @notice Sets the ERC-20 contract address (e.g., for USDC payments).
+    function setERC20ContractAddress(address contractAddress_) external override onlyOwner {
+        require(contractAddress_ != address(0), "Invalid ERC20 contract address");
+        erc20ContractAddress = contractAddress_;
+    }
+
+    /// @notice Gets the ERC-20 contract address used for payments.
+    function getERC20ContractAddress() external view override(IERC20Payable) returns (address) {
```

Comment:
Is `override(IERC20Payable)` needed here?

Reply:
removed